app/Services/UserService.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Services;
+
+use App\Http\Requests\Api\Auth\LoginRequest;
+use App\Http\Requests\Api\Auth\RegisterRequest;
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use App\Models\User;
+use App\Repositories\OfferRepository;
+use App\Repositories\StatusRepository;
+use App\Repositories\UserRepository;
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Hash;
+
+class UserService
```

Comment:
@aminetiyal I think separating each function into individual service will be cleaner. If you check the OfferService you will find only one method for executing the service, and this service is used for creating offers only and Rana will create another service for updating offer and will rename OfferService to CreatingOfferService. I am very sorry for the late review but the class has a lot of functions and will getting worse if we add the rest of the user auth. functions into it like forge/reset password.

Reply:
I took [OfferService](https://github.com/BuildForSDGCohort2/Team-248-Backend/blob/develop/app/Services/OfferService.php#L13) as an example, I noticed that the service name is OfferService and i guest more logic will be implemented inside.

[commit](https://github.com/BuildForSDGCohort2/Team-248-Backend/pull/29/commits/f6ae2804e5da0d4f5b762751e7e34dd6f41aa23e)

Reply:
I took [OfferService](https://github.com/BuildForSDGCohort2/Team-248-Backend/blob/develop/app/Services/OfferService.php#L13) as an example, I noticed that the service name is OfferService and guessed more logic will be implemented inside.

[commit](https://github.com/BuildForSDGCohort2/Team-248-Backend/pull/29/commits/f6ae2804e5da0d4f5b762751e7e34dd6f41aa23e)